File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/ShieldToUI.java
/**
 * @file ShieldToUI.java
 * @author Profesor
 */
package deepspace;

/**
 * @brief Representación ToUI de ShieldBooster
 */
public class ShieldToUI {
    private float boost;
    private int uses;

    /**
     * @brief Constructor con un parámetro
     * @param s Instancia de la clase ShieldBooster
     */
    ShieldToUI (ShieldBooster s) {
        boost=s.getBoost();
        uses=s.getUses();
    }    

    /**
     * @brief Consultor del valor del potenciador
     * @return boost
     */
    public float getBoost() {
        return boost;
    }

    /**
     * @brief Consultor de los usos
     * @return uses
     */
    public int getUses() {
        return uses;
    }
}

File: practicas/practica_2/Deepspace_Ruby_P2/lib/ShieldToUI.rb
#encoding:utf-8

# File: ShieldToUI.rb
# Author: Profesor

module Deepspace;

# Brief: Representación ToUI de ShieldBooster
class ShieldToUI 
  
  # Consultores
  attr_reader :boost, :uses

  # Brief: Constructor con un parámetro
   # Param s: Instancia de la clase ShieldBooster
  def initialize (s)
    @boost=s.boost
    @uses=s.uses
  end
  
  # Brief: Función para representar una instancia de la clase ShieldToUI
  #        en un string
  # Return: out String que representa una instancia de la clase ShieldToUI
  def to_s
    return "Boost: #{@boost}, Uses: #{@uses}"
  end
    
end # Class

end # Module